Add Blocks API and add additional Get document descriptions. (#7836)
opensearch-project/documentation-website: _api-reference/index-apis/blocks.md

---
layout: default
title: Blocks
parent: Index APIs
nav_order: 6
---

# Blocks
**Introduced 1.0**
{: .label .label-purple }

> The Blocks API limits which operations are available on the specified index. OpenSearch provides different types of blocks that allow you to restrict write, read, or metadata operations on an index. When adding a write block through the API, it ensures that all shards of the index have properly accounted for the block before returning a successful response to the user. This means that any in-flight write operations to the index have been completed before the write block takes effect.

Reviewer: ```suggestion
Use the Blocks API to limit certain operations on a specified index. The different types of blocks allow you to restrict write, read, or metadata operations on an index. 
For example, adding a `write` block through the API ensures that all shards of the index have properly accounted for the block before returning a successful response to the user. Any in-flight write operations to the index must have been completed before the `write` block takes effect.
```

Reviewer: I don't really understand why you are calling out the 'write' block here without describing the others. Can you accomplish the same thing by describing it in the table row?

Author: We do describe all of the options in the table row on line 25.

Reviewer: Yes, since they are in the table, then why call it out in the opening paragraph? If you describe one, you should describe them all. That's why I added the 'for example', because it seemed a bit out of place. But if the 'for example' works, then it should be fine to leave.

Author: For example is fine with me.

> The Blocks API limits which operations are available on the specified index. OpenSearch provides different types of blocks that allow you to restrict write, read, or metadata operations on an index.  When adding a write block through the API, it ensures that all shards of the index have properly accounted for the block before returning a successful response to the user. This means that any in-flight write operations to the index have been completed before the write block takes effect.

Author: ```suggestion
The Blocks API limits which operations are available on the specified index. OpenSearch provides different types of blocks that allow you to restrict write, read, or metadata operations on an index. When adding a write block through the API, it ensures that all shards of the index have properly accounted for the block before returning a successful response to the user. This means that any in-flight write operations to the index have been completed before the write block takes effect.
```

## Path and HTTP methods

```json
PUT /<index>/_block/<block>
```

## Path parameters

> | Parameter | Type | Description |

Reviewer: ```suggestion
| Parameter | Data type | Description |
```

---
layout: default
title: Blocks
parent: Index APIs
nav_order: 6
---

# Blocks
**Introduced 1.0**
{: .label .label-purple }

Use the Blocks API to limit certain operations on a specified index. The different types of blocks allow you to restrict write, read, or metadata operations on an index. 
For example, adding a `write` block through the API ensures that all shards of the index have properly accounted for the block before returning a successful response to the user. Any in-flight write operations to the index must have been completed before the `write` block takes effect.

## Path and HTTP methods

```json
PUT /<index>/_block/<block>
```

## Path parameters

| Parameter | Data type | Description |
:--- | :--- | :---
| `index` | String | A comma-delimited list of index names. Wildcard expressions (`*`) are supported. To target all data streams and indexes in a cluster, use `_all` or `*`. Optional. |

> | `<block>` | String | Specifies the type of block to apply to the index.  Valid values are: <br> `metadata`: Disables all metadata changes, such as closing the index. <br> `read`: Disables any read operations. <br> `read_only`: Disables any write operations and metadata changes. <br> `write`: Disables write operations. However, metadata changes are still allowed. |

Author: ```suggestion
| `<block>` | String | Specifies the type of block to apply to the index. Valid values are: <br> `metadata`: Disables all metadata changes, such as closing the index. <br> `read`: Disables any read operations. <br> `read_only`: Disables any write operations and metadata changes. <br> `write`: Disables write operations. However, metadata changes are still allowed. |
```
